code/game/objects/structures/tables_racks.dm

```diff
@@ -544,11 +544,12 @@
 	check_patient()
 
 /obj/structure/table/optable/proc/check_patient()
-	var/mob/living/carbon/human/M = locate(/mob/living/carbon/human, loc)
+	var/mob/living/carbon/M = locate(/mob/living/carbon) in loc
 	if(M)
 		if(M.resting)
 			patient = M
-			return TRUE
+			if(ishuman(M) ||  ismonkey(M))
+				return TRUE

```

Comment:
This logic is a bit odd, as it might set up the patient while returning a falsey value.

Reply:
That's the point. You can't output simplemob data to opcomps, but you will be able to perform researched surgeries on everything.

Reply:
While it makes sense given the context, it's pretty confusing logic.
This proc doubles as a human-or-monkey-patient check and a way to set the `patient` var.
Could you change it, or at least its name, for something more sane and clear?
Ideally you substitute it for a proc that checks, and handle the `patient` var assignment elsewhere.